diesel_derives/src/as_expression.rs

```diff
@@ -1,30 +1,30 @@
-use proc_macro2;
-use syn;
+use proc_macro2::TokenStream;
+use syn::DeriveInput;
 
-use meta::*;
-use util::*;
+use model::Model;
+use util::{ty_for_foreign_derive, wrap_in_dummy_mod};
 
-pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagnostic> {
-    let flags =
-        MetaItem::with_name(&item.attrs, "diesel").unwrap_or_else(|| MetaItem::empty("diesel"));
-    let is_sized = !flags.has_flag("not_sized");
+pub fn derive(item: DeriveInput) -> TokenStream {
+    let model = Model::from_item(&item, true);
 
-    let sql_types = MetaItem::all_with_name(&item.attrs, "sql_type");
-    let any_sql_types = !sql_types.is_empty();
-    let sql_types = sql_types
-        .into_iter()
-        .filter_map(|attr| attr.ty_value().map_err(Diagnostic::emit).ok());
+    if model.sql_types.is_empty() {
+        abort_call_site!(
```

Comment:
I think it's better if we error out in this case because we don't people (who generally ignore warnings) to forget to put sql type for this derive (which is what this derive is all about in the first place).

Reply:
:+1: